Add A basic guide to PHP Symfony (#38)
hetzneronline/community-content: tutorials/a-basic-guide-to-php-symfony/01.en.md

 create mode 100644 config/packages/swiftmailer.yaml
 create mode 100644 config/packages/test/framework.yaml
 create mode 100644 config/packages/test/monolog.yaml
 create mode 100644 config/packages/test/routing.yaml
 create mode 100644 config/packages/test/swiftmailer.yaml
 create mode 100644 config/packages/test/web_profiler.yaml
 create mode 100644 config/packages/translation.yaml
 create mode 100644 config/packages/twig.yaml
 create mode 100644 config/packages/validator.yaml
 create mode 100644 config/routes.yaml
 create mode 100644 config/routes/annotations.yaml
 create mode 100644 config/routes/dev/twig.yaml
 create mode 100644 config/routes/dev/web_profiler.yaml
 create mode 100644 config/services.yaml
 create mode 100644 phpunit.xml.dist
 create mode 100644 public/index.php
 create mode 100644 src/Controller/.gitignore
 create mode 100644 src/Entity/.gitignore
 create mode 100644 src/Kernel.php
 create mode 100644 src/Migrations/.gitignore
 create mode 100644 src/Repository/.gitignore
 create mode 100644 symfony.lock
 create mode 100644 templates/base.html.twig
 create mode 100644 tests/.gitignore
 create mode 100644 translations/.gitignore
```

> </details>

Reviewer: ```suggestion

</details>
```

---
SPDX-License-Identifier: MIT
path: "/tutorials/a-basic-guide-to-php-symfony-setup"
slug: "a-basic-guide-to-php-symfony-setup"
date: "2019-03-16"
title: "A basic guide to PHP Symfony - Setup"
short_description: "Symfony is a powerful PHP Framework. In this tutorial series we will take a look at many parts of it. In this part we are going to setup a new symfony project."
tags: ["Development", "Lang:PHP", "Lang:HTML", "Lang:Twig"]
author: "Moritz Fromm"
author_link: "https://github.com/frommMoritz/"
author_img: "https://avatars.githubusercontent.com/u/34239260"
author_description: "Manipulating arrays of characters in modern text editors that need more RAM than we used to do a flight to the moon. But it's super awesome... "
language: "en"
available_languages: ["en"]
header_img: ""
---

## Introduction

Symfony is a popular PHP framework.

> > A Framework is a set of components and functions that can be reused multiple times.

Reviewer 2: `Framework` should be lowercase.

> For example, it contains [Twig](https://twig.symfony.com/), which is a powerful templating engine, so the PHP code is separated from your HTML.

Reviewer 2: I would add `code` after the `HTML`... i think this sounds better

> This tutorial is split into multiple parts. In this one, we go over how to set up our Symfony project. We go over to learn more about Controllers and Routes. Afterward, we learn a bit about the `.env` File and start working with the database. Securing our application is crucial so we also take a look at authenticating users.

Reviewer 2: `controllers` and `routes` should be lowercase, as well as `file`

## Series index

1. Setup (You are here)
2. [Routes, Templates and Controllers](/tutorials/a-basic-guide-to-php-symfony-routes-templates-and-controllers)
3. _Working with the database (Not yet finished)_
4. _Security (Not yet finished)_

**Prerequisites**

Please note: Those prerequisites are needed at the time of writing. The specific version required, may change at any time.

> - PHP in version [7.1.3 or higher](https://github.com/symfony/symfony/blob/master/composer.json#L19).

Reviewer 2: Please drop the `in`.

  - A fair amount of PHP experience is recommended

> - [Composer](https://getcomposer.org/). You can check your PHP version by running `php -v`

Reviewer 2: I guess the sentence `You can check your PHP [...]` should not be here?

  - In this tutorial, we will assume that composer is installed globally and accessible via the command `composer`
  - Otherwise, you can replace the `composer` command with `php path/to/composer.phar`. [Read more...](https://getcomposer.org/doc/00-intro.md#locally)

> - A (simple) code editor, I am using [Visual Studio Code](https://code.visualstudio.com/)

Reviewer 2: Sometimes you use punctuation on this list and sometimes not, should not use punctuation.

This tutorial is tested on the following operating systems:

- Ubuntu 18.04.2 LTS bionic with PHP 7.2.15

## Step 1 - Create a new Symfony project

> We will start by creating a simple project. This is based on the `symfony/website-skeleton`. It provides an already configured Symfony application, so we will not have to configure it all on our own. The last argument passed is the folder to use. In our case, we use the name of this tutorial: `a-basic-guide-to-php-symfony`.

Reviewer 2: ```suggestion
We will start by creating a simple project. This is based on the `symfony/website-skeleton`. It provides an already configured Symfony application, so we don't have to configure everything ourselves. The last argument passed is the folder to use. In our case, we use the name of this tutorial: `a-basic-guide-to-php-symfony`.
```

Executing script cache:clear [OK]
Executing script assets:install public [OK]

Some files may have been created or updated to configure your new packages.
Please review, edit and commit them: these files are yours.

              
 What's next?
              

  * Run your application:
    1. Change to the project directory
    2. Create your code repository with the git init command
    3. Run composer require server --dev to install the development web server or configure another supported web server https://symfony.com/doc/current/setup/web_server_configuration.html

  * Read the documentation at https://symfony.com/doc

                        
 Database Configuration
                        

  * Modify your DATABASE_URL config in .env

  * Configure the driver (mysql) and
    server_version (5.7) in config/packages/doctrine.yaml

              
 How to test?
              

  * Write test cases in the tests/ folder
  * Run php bin/phpunit
```

</details>

Now we can `cd` (change directory) into the `a-basic-guide-to-php-symfony` folder. The rest of this tutorial will assume that we are in this directory.

> When you are developing "real" applications, using Git as a version control system is a great idea. You can initialize a repository and commit the initial state by the following command:

Reviewer 2: i would replace `by` with `with`

```bash
$ php bin/console server:run

                                                                                                                        
 [OK] Server listening on http://127.0.0.1:8000                                                                         
                                                                                                                        

 // Quit the server with CONTROL-C.

PHP 7.2.15-0ubuntu0.18.04.1 Development Server started at Fri Mar  8 22:23:14 2019
Listening on http://127.0.0.1:8000
Document root is /path/to/a-basic-guide-to-php-symfony/public
Press Ctrl-C to quit.

```

Please note, the address:port part (`[0.0.0.0:8080]`) is optional and is `the address to listen to.`

To keep the development server running in the background, the terminal multiplexer [tmux](https://www.hamvocke.com/blog/a-quick-and-easy-guide-to-tmux/) is a good option.

> Assuming you have not used a custom address port, we can now visit [http://localhost:8000/](http://localhost:8000/). There we are greeted by the "Welcome to Symfony 4.2.3" page containing a few simple pieces of information about the application. On the bottom, we find the profiler, a tool that comes in handy later when debugging. For now, it can safely be minimized by clicking the small `X` on the bottom right.

Reviewer 2: ```suggestion
Assuming you have not used a custom address port, we can now visit [http://localhost:8000/](http://localhost:8000/). There we are greeted by the "Welcome to Symfony 4.2.3" page containing a few information about the application. On the bottom we find the profiler, a tool that is useful for debugging later. For now, it can safely be minimized by clicking the small `X` on the bottom right.
```

## Step 3 - Exploring the file system

> Before we start creating our first "real" content, we explore the file system of Symfony a bit. The directory structure is like this:

Reviewer 2: directory structure != filesystem
```suggestion
Before we start creating our first "real" content, we explore the directory structure of Symfony a bit. The directory structure is like this:
